### ISSUE TITLE
yarn global add express-generator doesn't work

### ISSUE THREAD

**Issue by nguyendviet** [yarnpkg/yarn#5167]
```
$ yarn --version
1.3.2
```
**Do you want to request a *feature* or report a *bug*?**

bug
**What is the current behavior?**
```
$ yarn global add express-generator@4
yarn global v1.3.2
[1/4] Resolving packages...
[2/4] Fetching packages...
[3/4] Linking dependencies...
[4/4] Building fresh packages...
success Installed "express-generator@4.15.5" with binaries:
      - express
Done in 0.56s.

$ express
bash: express: command not found

$ yarn global add express-generator
yarn global v1.3.2
[1/4] Resolving packages...
[2/4] Fetching packages...
[3/4] Linking dependencies...
[4/4] Building fresh packages...
success Installed "express-generator@4.15.5" with binaries:
      - express
Done in 0.73s.

$ express
bash: express: command not found

$ express -h
bash: express: command not found

$ express test-app
bash: express: command not found

```

**If the current behavior is a bug, please provide the steps to reproduce.**

**What is the expected behavior?**

express command should be installed

**Please mention your node.js, yarn and operating system version.**
Windows 10
```
$ git --version
git version 2.13.2.windows.1
```

**Comment by rally25rs** [comment]
It looks like Yarn's global bin directory probably isn't on your `PATH`.
What method did you use to install Yarn?

**Comment by nguyendviet** [comment]
Hi Jeff, 

You're right. The bin directory isn't on my PATH. I installed yarn with
the yarn installer. I don't see any yarn guide mentioning setting PATH,
unless I missed it. 

So what I need to do now is: go to Control Panel > System > Advanced >
Environment Variables > Path > Edit, then paste the path of the global
yarn bin there below the existing paths, is that correct? 

If not, please send a guide to me so I can follow. Also, I've tried
installing express-generator with npm install and it just works. Why is
that? 

Cheers,

---
_VIET NGUYEN_ 

On 2018-01-07 11:37, Jeff Valore wrote:

> It looks like Yarn's global bin directory probably isn't on your PATH.
> What method did you use to install Yarn? 
> 
> --
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub [1], or mute the thread [2].
 

Links:
------
[1] https://github.com/yarnpkg/yarn/issues/5167#issuecomment-355834943
[2]

**Comment by rally25rs** [comment]
Yes that would be the correct action to take to set the path. There is a [step in the installer](https://github.com/yarnpkg/yarn/blob/master/resources/winsetup/Yarn.wxs#L44-L52) to do that, so not sure why it wouldn't have been set on your system.

> I've tried installing express-generator with npm install and it just works.

NPM's entry is probably already on your PATH.

-----

Assuming fixing the path works, is it OK to close this issue?

**Comment by nguyendviet** [comment]
Thank you for your answer. I've added yarn global bin to my PATH. however, I encounter this issue:
```
$ yarn global bin
C:\Users\{user name}\scoop\apps\yarn\current\bin
```

added to PATH. then:
```
$ yarn global add express-generator
yarn global v1.3.2
[1/4] Resolving packages...
[2/4] Fetching packages...
[3/4] Linking dependencies...
[4/4] Building fresh packages...
success Installed "express-generator@4.15.5" with binaries:
      - express
Done in 0.62s

$ express
/bin/sh: /c/Users/{user name}/scoop/apps/yarn/current/bin/../../../../../AppData/Local/Yarn/config/global/node_modules/.bin/express: No such file or directory
```

**Comment by bluelovers** [comment]
i have same issue too
install by msi and npm i -g yarn, both not work well

---

`npm global` install bin in `C:\Users\USER\AppData\Roaming\npm`
`yarn global` install bin in `C:\Users\USER\AppData\Roaming\npm\bin`

the `PATH` env didn't has `C:\Users\USER\AppData\Roaming\npm\bin`

is `C:\Users\USER\AppData\Roaming\npm\bin` path didn't add to PATH env 
or bin path is wrong, should same as `C:\Users\USER\AppData\Roaming\npm`

**Comment by palsdm** [comment]
Hi @rally25rs 
I am running Windows 10.

Tried to install Yarn via:
- normal cmd: does not add the "...\Users\...\AppData\Local\Yarn\bin" to the PATH env var
- cmd as administrator: "npm i -g yarn" hangs on "[       ...........] \ extract:yarn: verb lock using ..."

Once I manually add "...\Users\...\AppData\Local\Yarn\bin" to the PATH env var not only express-generator works (it was put there all along) but running "npm i -g yarn" does not hang anymore.

There must be an issue somewhere between ...\AppData\Roaming\... and ...\AppData\Local\...

Note: running versions npm 6.14.11 and yarn 1.22.10
